docs/source/developers/python.rst

```diff
@@ -518,13 +518,11 @@ We set a number of environment variables:
 
 - the path of the installation directory of the Arrow C++ libraries as
   ``ARROW_HOME``
-- add the path of installed DLL libraries to ``PATH``
 - and the CMake generator to be used as ``PYARROW_CMAKE_GENERATOR``
 
 .. code-block::
 
-   $ set ARROW_HOME=%cd%\arrow-dist
-   $ set PATH=%ARROW_HOME%\bin;%PATH%
+   $ set ARROW_HOME=%CONDA_PREFIX%
```

Comment:
The Appveyor build script is using `%CONDA_PREFIX%\Library`, so it would be good to verify if that "Library" part is needed.

https://github.com/apache/arrow/blob/a1d7a449295adcc1f407d7b0037fbc308492372a/ci/appveyor-cpp-build.bat#L123

Reply:
I tested locally with `ARROW_HOME=%CONDA_PREFIX%` and it worked without an issue. But could change to `%CONDA_PREFIX%\Library` if preferred.

Reply:
So locally it works for you with both options?

Reply:
Haven't tested `ARROW_HOME=%CONDA_PREFIX%\Library` but did built with `ARROW_HOME=%CONDA_PREFIX%`. Should I test both?

Reply:
Hmm, perhaps `ARROW_HOME` is simply not necessary? I just tested without it on Linux and it worked.

Reply:
I guess `CMAKE_PREFIX_PATH` should be added instead?

Reply:
I don't know, is it actually required?

Reply:
Hm, not sure. Will look around the cmake changes tomorrow.

Reply:
I think `ARROW_HOME` is still needed:
https://github.com/apache/arrow/blob/21dbf4ac09583651899232c9a80b3f5d6580a588/python/pyarrow/src/CMakeLists.txt#L43-L44
Maybe it somehow works in conda env but it didn't work for me (MacOS with `brew` and `pip`):
```
(pyarrow-dev-9) (base) alenkafrim@Alenkas-MacBook-Pro arrow % arrow cpp
~/repos/arrow/cpp/build ~/repos/arrow
[585/585] Linking CXX executable debug/arrow-dataset-scanner-test
[1/2] Install the project...
-- Install configuration: "DEBUG"
CMake Error at cmake_install.cmake:41 (file):
  file cannot create directory: /lib/cmake/Arrow.  Maybe need administrative
  privileges.


FAILED: CMakeFiles/install.util 
cd /Users/alenkafrim/repos/arrow/cpp/build && /opt/homebrew/Cellar/cmake/3.24.2/bin/cmake -P cmake_install.cmake
ninja: build stopped: subcommand failed.
~/repos/arrow
```

after exporting `ARROW_HOME` the installation was successful.

Reply:
`ARROW_HOME` is also "necessary" if you follow the developers docs because we use this env variable to set some other things, like `CMAKE_INSTALL_PREFIX`. But you can of course pass a path to that as well, without using ARROW_HOME for this.

And on unix, I suppose having ARROW_HOME defined so that cmake adds it to `CMAKE_PREFIX_PATH` might not be needed when using cmake from conda (then the conda prefix might already be included in ``CMAKE_SYSTEM_PREFIX_PATH``

Reply:
But given that our appveyor setup actually uses `%CONDA_PREFIX%\Library`, let's keep this consistent with that? 
Also in windows.rst for cpp,  `%CONDA_PREFIX%\Library` is mentioned.  
According to https://docs.conda.io/projects/conda-build/en/stable/user-guide/environment-variables.html#environment-variables-set-during-the-build-process, conda uses a `Library` subdirectory in the conda env.

Reply:
Ok, will correct and make a test locally, just to be sure.